net/kernel_socket.cpp

```diff
@@ -99,8 +84,10 @@ class KernelSocketStream : public SocketStreamBase {
     }
     ~KernelSocketStream() override {
         if (fd < 0) return;
+        auto errno_bak = errno;
         shutdown(ShutdownHow::ReadWrite);
         close();
+        errno = errno_bak;
```

Comment:
backup the errno, otherwise it will be overwritten when `KernelSocketStream` destruct in `KernelSocketClient::do_connect`

Reply:
It's not always needed to save and restore errno in this destructor. We'd better do it in the context that needs it.